Decouple main modules from event store modules. (#93)
hallgren/eventsourcing: README.md

}
```

The aggregate needs to implement the `Transition(event eventsourcing.Event)` and `Regsiter(r eventsourcing.RegsiterFunc)` methods to fulfill the aggregate interface. This methods define how events are transformed to build the aggregate state and which events to register into the repository. 

Example of the Transition method from the `Person` aggregate.

```go
// Transition the person state dependent on the events
func (person *Person) Transition(event eventsourcing.Event) {
    switch e := event.Data().(type) {
    case *Born:
            person.Age = 0
            person.Name = e.Name
    case *AgedOneYear:
            person.Age += 1
    }
}
```

The `Born` event sets the `Person` property `Age` and `Name`, and the `AgedOneYear` adds one year to the `Age` property. This makes the state of the aggregate flexible and could easily change in the future if required.

> Exampel or the Register method.

Reviewer: ```suggestion
Example or the Register method:
```

}
```

Internally the `TrackChange` methods calls the `Transition` method on the aggregate to transform the aggregate based on the newly created event.

To bind metadata to events use the `TrackChangeWithMetadata` method.
  
The internal `Event` looks like this.

```go
type Event struct {
    // aggregate identifier 
    aggregateID string
    // the aggregate version when this event was created
    version         Version
    // the global version is based on all events (this value is only set after the event is saved to the event store) 
    globalVersion   Version
    // aggregate type (Person in the example above)
    aggregateType   string
    // UTC time when the event was created  
    timestamp       time.Time
    // the specific event data specified in the application (Born{}, AgedOneYear{})
    data            interface{}
    // data that don´t belongs to the application state (could be correlation id or other request references)
    metadata        map[string]interface{}
}
```

> To access properties on the event there are methods exposing them. This prevent external parties to modify the event from the outside.
> 
> ```go
>     func (e Event) AggregateID() string {
>         return e.event.AggregateID
>     }
> 
>     func (e Event) Version() Version {
>         return Version(e.event.Version)
>     }
> 
>     func (e Event) GlobalVersion() Version {
>         return Version(e.event.GlobalVersion)
>     }
> 
>     func (e Event) AggregateType() string {
>         return e.event.AggregateType
>     }
> 
>     func (e Event) Timestamp() time.Time {
>         return e.event.Timestamp
>     }
> 
>     func (e Event) Data() interface{} {
>         return e.data
>     }
> 
>     func (e Event) Metadata() map[string]interface{} {
>         return e.metadata
>     }
> 
>     // Reason is based on the data type
>     func (e Event) Reason() string {
>         if e.data == nil {
>             return ""
>         }
>         return reflect.TypeOf(e.data).Elem().Name()
>     }
> ```
> 

Reviewer: I think it's preferable if you could simply mention that there exists methods for accessing these fields, instead of having a copy of the actual code in the documentation. If you change the code, you should also update this.

```suggestion
To access properties on the event you can use the corresponding methods exposing them, e.g `AggregateID()`. This prevent external parties to modify the event from the outside.
```

## Custom made components

> Parts of this package may not fulfill your application need, either it can be that the event store uses the wrong database for storage.

Reviewer: Can a database be wrong, or maybe there doesn't exists an adapter for the database that you use?
```suggestion
Parts of this package may not fulfill your application need, for instance an event store adapter for your database of choice.
```
